tests/zenko_tests/node_tests/iam_policies/cloudserver/IAMUser.js

```diff
@@ -0,0 +1,257 @@
+            userCredentials.secretAccessKey = res[3].AccessKey.SecretAccessKey;
+            done();
+        });
+    });
+
+    after(done => {
+        async.series([
+            next => iam.deleteUser({UserName: userName}, next),
+            next => s3.deleteBucket({Bucket: bucketName}, next),
+            next => s3.deleteBucket({Bucket: bucketName2}, next),
+        ], err => {
+            assert.ifError(err);
+            done();
+        });
+    })
+
+    it('should not be able to metadata search without the policy s3:MetadataSearch', done => {
+        return metadataSearchResponseCode(userCredentials, bucketName, (err, res) => {
+            if (err) {
+                assert.ifError(err)
+                return done(err)
+            }
+            assert.strictEqual(res, 403);
+            return done();
+        });
+    });
+
+    it('should be able to metadata search with the policy s3:MetadataSearch allowed', done => {
+        async.series([
+            next => iam.createPolicy({PolicyName: allowMdSearchPolicyName, PolicyDocument: createPolicy("s3:MetadataSearch")}, (err, res) => {
+                if (err)
+                    return next(err);
+                return iam.attachUserPolicy({UserName: userName, PolicyArn: res.Policy.Arn}, next);
+            }),
+            next => metadataSearchResponseCode(userCredentials, bucketName, (err, res) => {
+                    if (err) {
+                        assert.ifError(err)
+                        return next(err)
+                    }
+                    assert.strictEqual(res, 200);
+                    return next();
+                }),
+            next => iam.listPolicies({}, (err, res) => {
+                    if (err) {
+                        return next(err);
+                    }
+                    const p = res.Policies.find(p => p.PolicyName === allowMdSearchPolicyName);
+                    if (p)
+                        return iam.detachUserPolicy({UserName: userName, PolicyArn: p.Arn}, (err, res) => {
+                            if (err)
+                                return next(err);
+                            return iam.deletePolicy({PolicyArn: p.Arn}, next);
+                        })
+                    return next();
+                }),
+            ],
+            err => {
+                assert.ifError(err);
+                done();
+            })
+    });
+
+    it('should be able to metadata search with the policy s3:MetadataSearch allowed' +
+        'on specific bucket but not on others', done => {
+        async.series([
+                next => iam.createPolicy({PolicyName: allowMdSearchPolicyName, PolicyDocument: createPolicy("s3:MetadataSearch", true, `arn:aws:s3:::${bucketName}`)}, (err, res) => {
+                    if (err)
+                        return next(err);
+                    return iam.attachUserPolicy({UserName: userName, PolicyArn: res.Policy.Arn}, next);
```

Comment:
I would suggest creating a null variable `policyArn` outside of async.series() and assign `res.Policy.Arn` to it here.
Then in line 142, we don't need to `listPolicies`, we can directly do the `detachUserPolicy` and `deletePolicy` using the `policyArn` saved before